buffer/src/main/java/io/netty/buffer/api/MemoryManager.java

```diff
@@ -47,6 +49,20 @@ static MemoryManager instance() {
         return MemoryManagerOverride.configuredOrDefaultManager();
     }
 
+    /**
+     * Register a callback that will be called whenever a {@link Buffer} instance is leaked.
+     * <p>
+     * Be mindful that the callback must be fast, and not take any locks or call any blocking methods,
+     * as this might interfere with the garbage collectors reference processing and cleaning.
```

Comment:
Should we also add some hint that if you use logging you want to use async logging due the reasons stated above ?